elementary/monitor/data_monitoring/alerts/data_monitoring_alerts.py

```diff
@@ -0,0 +1,349 @@
+from elementary.monitor.alerts.model_alert import ModelAlertModel
+from elementary.monitor.alerts.source_freshness_alert import SourceFreshnessAlertModel
+from elementary.monitor.alerts.test_alert import TestAlertModel
+from elementary.monitor.api.alerts.alerts import AlertsAPI
+from elementary.monitor.api.alerts.schema import AlertsSchema
+from elementary.monitor.data_monitoring.alerts.integrations.slack.slack import (
+    SlackIntegration,
+)
+from elementary.monitor.data_monitoring.data_monitoring import DataMonitoring
+from elementary.monitor.data_monitoring.schema import ALERT_TABLES
+from elementary.monitor.fetchers.alerts.schema import (
+    PendingModelAlertSchema,
+    PendingSourceFreshnessAlertSchema,
+    PendingTestAlertSchema,
+)
+from elementary.tracking.tracking_interface import Tracking
+from elementary.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class DataMonitoringAlerts(DataMonitoring):
+    def __init__(
+        self,
+        config: Config,
+        tracking: Optional[Tracking] = None,
+        filter: Optional[str] = None,
+        force_update_dbt_package: bool = False,
+        disable_samples: bool = False,
+        send_test_message_on_success: bool = False,
+        global_suppression_interval: int = 0,
+        override_config: bool = False,
+    ):
+        super().__init__(
+            config, tracking, force_update_dbt_package, disable_samples, filter
+        )
+
+        self.global_suppression_interval = global_suppression_interval
+        self.override_config = override_config
+        self.alerts_api = AlertsAPI(
+            self.internal_dbt_runner,
+            self.config,
+            self.elementary_database_and_schema,
+            self.global_suppression_interval,
+            self.override_config,
+        )
+        self.sent_alert_count = 0
+        self.send_test_message_on_success = send_test_message_on_success
+        self.override_meta_slack_channel = override_config
+        self.alerts_integraion = self._get_integration_client()
+
+    def _get_integration_client(self):
+        return SlackIntegration(config=self.config, tracking=self.tracking)
+
+    def _get_integration_params(
+        self,
+        alert: Union[
+            PendingTestAlertSchema,
+            PendingModelAlertSchema,
+            PendingSourceFreshnessAlertSchema,
+        ],
+    ) -> Dict[str, Any]:
+        return dict(
+            channel=(
+                self.config.slack_channel_name
+                if self.override_meta_slack_channel
+                else (alert.alert_channel or self.config.slack_channel_name)
+            )
+        )
+
+    def _fetch_data(self, days_back: int) -> AlertsSchema:
+        return self.alerts_api.get_new_alerts(
+            days_back=days_back,
+            disable_samples=self.disable_samples,
+            filter=self.filter.get_filter(),
+        )
+
+    def _format_alert(
```

Comment:
In my philosophy of code, those are 2 different class (although they are overlapping).
One (Pending alerts schema) is a schema which is used to validate typing and the structure of the data we query from the db using the alerts fetcher.
This will not be used in the SaaS version because we are going to query it from other tables and the fields probably going to be different.

The Alerts models are the "normalized" schema of alerts that is used by the integrations. This is the data structure we "sign" on that is going to contain all the data an integration need on an alert.

Merging the 2 is not a good option IMO because they have a different purpose and use, they first one is changeable depends on the system, and currently they are overlapping but they can easily be changed and don't 🙂

Reply:
I will move it 🙂 
This is scares me a bit because it can create a couple between the 2 in the future (I really like to separate the 2 - fetcher should not be aware of the API that is calling it, and adding the format to the schema kinda makes it coupled).
So if I will see in the future that we are coupling them I will ask to revert it to that change 🙂